### PR TITLE
New Trainer Initialization Interface

### PR DESCRIPTION
**Pull request by FengZiYjun**
- add Loss, Optimizer
Loss and Optimizer are arguments passed to Trainer.
```Python
    trainer = ClassificationTrainer(epochs=train_args["epochs"],
                                    batch_size=train_args["batch_size"],
                                    validate=train_args["validate"],
                                    use_cuda=train_args["use_cuda"],
                                    pickle_path=save_dir,
                                    save_best_dev=train_args["save_best_dev"],
                                    model_name=model_name,
                                    loss=Loss("cross_entropy"),
                                    optimizer=Optimizer("SGD", lr=0.001, momentum=0.9))
```
- change Trainer & Tester initialization interface: two styles of definition provided

```Python
    # Trainer: two definition styles
    # 1
    # trainer = SeqLabelTrainer(trainer_args)  # trainer_args is a dict

    # 2
    trainer = SeqLabelTrainer(
        epochs=trainer_args["epochs"],
        batch_size=trainer_args["batch_size"],
        validate=trainer_args["validate"],
        use_cuda=trainer_args["use_cuda"],
        pickle_path=pickle_path,
        save_best_dev=trainer_args["save_best_dev"],
        model_name=model_name,
        optimizer=Optimizer("SGD", lr=0.01, momentum=0.9),
    )

    # Model
    model = SeqLabeling(model_args)

    # Start training
    trainer.train(model, data_train, data_dev)
    print("Training finished!")

    # Tester
    tester = SeqLabelTester(save_output=False,
                            save_loss=False,
                            save_best_dev=False,
                            batch_size=8,
                            use_cuda=False,
                            pickle_path=pickle_path,
                            model_name="seq_label_in_test.pkl",
                            print_every_step=1
                            )

    # Start testing with validation data
    tester.test(model, data_test)
```
- handle Optimizer construction and loss function definition in a hard manner
Loss borrows functions from PyTorch. Optimizer is actually constructed inside Trainer. 
- add argparse in task-specific scripts. (seq_labeling.py & text_classify.py)
```Python
parser.add_argument("-s", "--save", type=str, default="./test_classification/", help="path to save pickle files")
parser.add_argument("-t", "--train", type=str, default="./data_for_tests/text_classify.txt",
                    help="path to the training data")
parser.add_argument("-c", "--config", type=str, default="./data_for_tests/config", help="path to the config file")
parser.add_argument("-m", "--model_name", type=str, default="classify_model.pkl", help="the name of the model")
```
- seq_labeling.py & text_classify.py work

### CODE REVIEW

**Comment by xpqiu** [comment]
good

**Comment by xuyige** [comment]
so "-s xxx" is equal to "--save xxx" ?

**Comment by FengZiYjun** [comment]
@xuyige yes
